post-repo-api/src/main/java/klon/post/repo/api/PostRepository.java

```diff
@@ -0,0 +1,23 @@
+package klon.post.repo.api;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public interface PostRepository {
+
+    void addPost(Post post);
+
+    Optional<Post> getPost(String postId);
+
+    Stream<Post> getPostsByUser(String userId);
+
+    /**
+     * Retrieves post for User starting with the Post having passed post id and no older than that.
```

Comment:
Nie wiem czy dobrze rozumiem, ale czy tutaj jest sortowanie od najstarszego do najnowszego ("no older that that")? Jeśli tak, to chyba trzeba to odwrócić. Jeśli mamy wyświetlać na "wallu" czyjeś posty, to idziemy od najnowszego do najstarszego. Tak to działa w "socialach"
W użytkowego punktu widzenia wydaje mi się też, że Post z podanym identyfikatorem powinien być odfiltrowany. Bierzesz pierwsze 10 ze streama i jeśli później chcesz wziąć kolejne, to znasz ostatni, który pobrałeś i nie chcesz go dostać kolejny raz.

Reply:
Faktycznie, chyba nie powinienem takich rzeczy robić z samego rana przed kawą... W tym momencie to zadziała tak, że mając 3 elementy w streamie i wskazując środkowego otrzymamy tylko tego z najstarszą datą. Wydaje mi się, że teraz to już będzie okej.

Reply:
No to tylko update javadoca. Jakoś tak:
- returns in descending order based on the date
- starting from the next after given post